src/leader.py

```diff
@@ -62,12 +62,17 @@ def turn_right(self):
     def turn_straight(self):
         self.fw.turn(self.STRAIGHT_ANGLE)
 
-
-    def turn(self,position):
-        if position == -1:
-            self.turn_left()
-        elif position == 1:
-            self.turn_right()
+    def set_turn_angle(self,position):
+        self.fw.turn(self.STRAIGHT_ANGLE + 45  * position)
+    
```

Comment:
According to PEP8 style guide, you need to have two blank lines between functions.